### PR TITLE
Refactor mvcc properties collector

### PR DESCRIPTION
**Pull request by huachaohuang**
This PR refactors the mvcc properties and collector so we can add other properties and collectors more easily.

### CODE REVIEW

File: src/storage/mvcc/reader.rs [hunk 1]
```diff
@@ -514,11 +512,34 @@ impl<'a> MvccReader<'a> {
...
+        }
+        // Aggregate MVCC properties.
+        let mut props = MvccProperties::new();
+        for (_, v) in &*collection {
+            let mvcc = match MvccProperties::decode(v.user_collected_properties()) {
+                Ok(v) => v,
+                Err(_) => return None,
+            };
+            // Filter out properties after safe_point.
+            if mvcc.min_ts > safe_point {
```
Comment (zhangjinpeng1987) [inline src/storage/mvcc/reader.rs]: It is better to extract a Filter.

Comment (huachaohuang) [inline src/storage/mvcc/reader.rs]: I think we should keep it simple until we know what we need.
We can refactor it easily in the future.

**Comment by huachaohuang** [comment]
PTAL @disksing

**Comment by disksing** [comment]
LGTM

**Pull request merged**